runtime/compiler/optimizer/abstractinterpreter/AbsInterpreter.cpp

```diff
@@ -0,0 +1,3072 @@
+         state->push(value1);
+         state->push(value2);
+         break;
+         }
+      case TR::Float:
+         {
+         TR::AbsValue* value = createTopFloat();
+         state->push(value);
+         break;
+         }
+      case TR::Int8:
+      case TR::Int16:
+      case TR::Int32:
+         {
+         TR::AbsValue *value = createTopInt();
+         state->push(value);
+         break;
+         }
+      case TR::Int64:
+         {
+         TR::AbsValue *value1 = createTopLong();
+         TR::AbsValue *value2 = createTopLong();
+         state->push(value1);
+         state->push(value2);
+         break;
+         }
+      case TR::Address:
+         {
+         TR::AbsValue* value = createTopObject();
+         state->push(value);
+         break;
+         }
+      default:
+         TR_ASSERT_FATAL(false, "Invalid type");
+         break;
+      }
+   }
+
+void TR::AbsBlockInterpreter::arrayStore(TR::DataType type)
+   {
+   TR::AbsStackMachineState* state = getState();
+
+   if (type.isDouble() || type.isInt64())
+      state->pop(); //dummy
+
+   TR::AbsValue* value = state->pop();
+   TR_ASSERT_FATAL(type == TR::Int8 || type == TR::Int16 ? value->getDataType() == TR::Int32 : value->getDataType() == type, "Unexpected type");
+
+   TR::AbsValue *index = state->pop();
+   TR_ASSERT_FATAL(index->getDataType() == TR::Int32, "Unexpected type");
+
+   TR::AbsValue *arrayRef = state->pop();
+   TR_ASSERT_FATAL(arrayRef->getDataType() == TR::Address, "Unexpected type");
+
+   //heap is being not modeled
+   switch (type)
+      {
+      case TR::Double:
+      case TR::Float:
+      case TR::Int8:
+      case TR::Int16:
+      case TR::Int32:
+      case TR::Int64:
+      case TR::Address:
+         break;
+      default:
+         TR_ASSERT_FATAL(false, "Invalid type");
+         break;
+      }
+   }
+
+void TR::AbsBlockInterpreter::binaryOperation(TR::DataType type, BinaryOperator op)
+   {
+   TR::AbsStackMachineState* state = getState();
+
+   if (type.isDouble() || type.isInt64())
+      state->pop(); //dummy
+
+   TR::AbsValue* value2 = state->pop();
+   TR_ASSERT_FATAL(value2->getDataType() == type, "Unexpected type");
+
+   if (type.isDouble() || type.isInt64())
+      state->pop(); //dummy
+
+   TR::AbsValue* value1 = state->pop();
+   TR_ASSERT_FATAL(value1->getDataType() == type, "Unexpected type");
+
+   switch (type)
+      {
+      // float and double are not modeled
+      case TR::Float:
+         {
+         TR::AbsValue *result = createTopFloat();
+         state->push(result);
+         break;
+         }
+
+      case TR::Double:
+         {
+         TR::AbsValue *result1 = createTopDouble();
+         TR::AbsValue *result2 = createTopDouble();
+         state->push(result1);
+         state->push(result2);
+         break;
+         }
+
+      //The following types are modeled
+      case TR::Int32:
+         {
+         if (isIntConst(value1) && isIntConst(value2) && !insideLoop()) //both int const
```

Comment:
This is related to https://github.com/eclipse-openj9/openj9/pull/17813#discussion_r1427426335.

As you can see, if the block is inside loop. rather than setting the state that inherited from interpreted predecessor to top, which is a conservative way, we kept it as it is. I think this is becuase we think most instructions are atomic and missing inheriting states from some uninterpreted predecessors won't affect the soundness. Instead, we only introduce top when we interpret those arithmetic instruction which we think their soundness will be affected by missing those uninterpreted predcessors.